### PR TITLE
[MTSRE-870] creating backplane permissions for openshift-connectors to perform OLM dance

### PR DESCRIPTION
**Pull request by Ankit152**
Signed-off-by: Ankit Kurmi <akurmi@redhat.com>

Jira: [MTSRE-870](https://issues.redhat.com/browse/MTSRE-870)

### CODE REVIEW

File: deploy/backplane/mtsre/openshift-connectors/01-openshift-connectors-cr-admins.ClusterRole.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: backplane-openshift-connectors
```
Comment (yashvardhan-kukreja) [inline deploy/backplane/mtsre/openshift-connectors/01-openshift-connectors-cr-admins.ClusterRole.yml:4]:
```suggestion
  name: backplane-openshift-connectors-cr-admins
```

File: deploy/backplane/mtsre/openshift-connectors/10-mtsre-openshift-connectors-cr-admins.SubjectPermissions.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+apiVersion: managed.openshift.io/v1alpha1
```
Comment (yashvardhan-kukreja) [inline deploy/backplane/mtsre/openshift-connectors/10-mtsre-openshift-connectors-cr-admins.SubjectPermissions.yml:1]: Can you rename this file to `10-mtsre-openshift-connectors-admins.SubjectPermissions.yml`

File: deploy/backplane/mtsre/openshift-connectors/01-openshift-connectors-cr-admins.ClusterRole.yml (hunk 1 above)
Comment (Ankit152) [reply deploy/backplane/mtsre/openshift-connectors/01-openshift-connectors-cr-admins.ClusterRole.yml]: done

File: deploy/backplane/mtsre/openshift-connectors/10-mtsre-openshift-connectors-cr-admins.SubjectPermissions.yml (hunk 2 above)
Comment (Ankit152) [reply deploy/backplane/mtsre/openshift-connectors/10-mtsre-openshift-connectors-cr-admins.SubjectPermissions.yml]: done

**Comment by cblecker** [comment]
These are extremely high permissions. Is there a reason the existing backplane managed script can't be used?

**Comment by Ankit152** [comment]
At the present moment few of the `connectors-operator` installations in prod are stuck in older version. To fix that, we need to perform OLM dance and hence we require such permissions.

**Comment by cblecker** [comment]
@Ankit152 see https://github.com/openshift/managed-scripts/tree/main/scripts/SREP/delete-olm-operator

**Comment by yashvardhan-kukreja** [comment]
@cblecker thanks for pointing out the script. I wasn't aware of its existence, though I have the following questions around it:
- I do not see it listed in under the cluster where we are trying to do this OLM dance
```
~                                                                                                          20s 11:16:25
❯ ocm backplane scripts list                          
NAME                             DESCRIPTION
SREP/cluster-operator-status     Retrieves status of the cluster's Cluster Operators
SREP/example                     Example script
SREP/get-targets-down            Lists pods marked as down by cluster Prometheus
SREP/node-logs                   Read the node logs. Restrict specific logs of oc adm node-logs.
SREP/tsdb-status                 Get status of prom's time series DB
CEE/check-tech-preview-features  Checks if any tech preview feature sets are enabled. If enabled, lists feature set(s...
CEE/etcd-health-check            Prints out the etcd health info.
CEE/get-rhoc-connector-info      Collect information for the specific conenctor instance....
CEE/list-alerts                  Lists the active/pending alerts in alertmanager. It also provide an option to list a...
```
This is staging env and my backplane version is 0.0.33
- Am I right to assume that this managed-script during its execution would inherit only [these](https://github.com/openshift/managed-scripts/blob/main/scripts/SREP/delete-olm-operator/metadata.yaml#L9-L24) roles and would not be restricted in any way by our (MT-SRE's) backplane privileges?

**Comment by yashvardhan-kukreja** [comment]
> These are extremely high permissions.

Also @cblecker , just mentioning here so as to ensure that we all are on the same page. We're ultimately scoping this clusterRole to the redhat-openshift-connectors namespace only.
So would it be a problem where to still have these permissions? Because from my personal experience, there were instances, with addons like ocs-*, where we didn't necessarily had to perform the OLM dance but to tweak certain things in the CSV or delete just the CSV corresponding to a specific bundle/subscription. 
This kind of selective control does not seem to be possible with the backplane script you suggested - it seems to delete all the CSVs/Subscriptions/OperatorGroups and recreating them, which might be useful in a full-blown OLM dance but not other possible scenarios.

wdyt :)

**Comment by yashvardhan-kukreja** [comment]
Also, @cblecker, thanks for pointing out the script. I wasn't aware of its existence, though cutting-to-the-chase I have the following technical questions around it out of my curiosity:
- I do not see it listed in under the cluster where we are trying to do this OLM dance
```
~                                                                                                          20s 11:16:25
❯ ocm backplane scripts list                          
NAME                             DESCRIPTION
SREP/cluster-operator-status     Retrieves status of the cluster's Cluster Operators
SREP/example                     Example script
SREP/get-targets-down            Lists pods marked as down by cluster Prometheus
SREP/node-logs                   Read the node logs. Restrict specific logs of oc adm node-logs.
SREP/tsdb-status                 Get status of prom's time series DB
CEE/check-tech-preview-features  Checks if any tech preview feature sets are enabled. If enabled, lists feature set(s...
CEE/etcd-health-check            Prints out the etcd health info.
CEE/get-rhoc-connector-info      Collect information for the specific conenctor instance....
CEE/list-alerts                  Lists the active/pending alerts in alertmanager. It also provide an option to list a...
```
This is staging env and my backplane version is 0.0.33
- Am I right to assume that this managed-script during its execution would inherit only [these](https://github.com/openshift/managed-scripts/blob/main/scripts/SREP/delete-olm-operator/metadata.yaml#L9-L24) roles and would not be restricted in any way by our (MT-SRE's) backplane privileges?

**Comment by Ankit152** [comment]
@yashvardhan-kukreja attaching the PR that updates the group. PTAL: https://github.com/openshift/managed-scripts/pull/101

**Comment by yashvardhan-kukreja** [comment]
@cblecker mind taking a look at this [comment](https://github.com/openshift/managed-cluster-config/pull/1402#issuecomment-1373184597)?

**Comment by cblecker** [comment]
is this still needed if https://github.com/openshift/managed-scripts/pull/101 was done

**Comment by yashvardhan-kukreja** [comment]
@cblecker after performing a manual dance for another addon, it was found out that the script being used https://github.com/openshift/managed-scripts/pull/101 is not fully functional. 

> olm-operator and catalog-operator pods need to be restarted as a last step as well

Rest we'd still strongly prefer to have the permissions which are intended in this PR because of the following reason (taken from my last comment)

> I ask this because from my personal experience, there were instances, with addons like ocs-*, where we didn't necessarily had to perform the OLM dance but to tweak certain things in the CSV or delete just the CSV corresponding to a specific bundle/subscription.
This kind of selective control does not seem to be possible with the backplane script you suggested - it seems to delete all the CSVs/Subscriptions/OperatorGroups and recreating them, which might be useful in a full-blown OLM dance but not with other possible scenarios.

Feel free to let me know your opinions.

File: deploy/backplane/mtsre/openshift-connectors/10-mtsre-openshift-connectors-admins.SubjectPermissions.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+apiVersion: managed.openshift.io/v1alpha1
+kind: SubjectPermission
+metadata:
+  name: backplane-mtsre-openshift-connectors-admins
+  namespace: openshift-rbac-permissions
+spec:
+  permissions:
+  - allowFirst: true
+    clusterRoleName: admin
```
Comment (mrbarge) [inline deploy/backplane/mtsre/openshift-connectors/10-mtsre-openshift-connectors-admins.SubjectPermissions.yml:9]: Should this be `backplane-openshift-connectors-cr-admins` ? I note the ClusterRole is being defined but nothing seems to be using it.

Comment (yashvardhan-kukreja) [reply deploy/backplane/mtsre/openshift-connectors/10-mtsre-openshift-connectors-admins.SubjectPermissions.yml:9]: Ahh nice catch,
Actually, we'd want both `admin` as well as `backplane-openshift-connectors-cr-admins` to be associated here.
`admin` being the baseline admin permissions suitable for mt-sre team.
`backplane-openshift-connectors-cr-admins` being the extra set of required permissions not covered by the `admin` role.

@Ankit152 can you add 
```
  - allowFirst: true
    clusterRoleName: backplane-openshift-connectors-cr-admins
    namespacesAllowedRegex: (^redhat-openshift-connectors$)
```
as well under the `permissions` section

Comment (Ankit152) [reply deploy/backplane/mtsre/openshift-connectors/10-mtsre-openshift-connectors-admins.SubjectPermissions.yml:9]: In that case, I am adding `backplane-mtsre-monitoring` as well. 🙂

**Comment by jewzaam** [comment]
/approve

Synced with @cblecker on this and while we don't particularlly like the permissions they are very targeted and contained.

**Pull request merged**